File: SumaMatrices.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author Monica Ileana
 */
import java.util.Scanner;

public class SumaMatrices {
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.print("Ingrese el número de filas: ");
        int filas = scanner.nextInt();
        
        System.out.print("Ingrese el número de columnas: ");
        int columnas = scanner.nextInt();
        
        int[][] matrizA = new int[filas][columnas];
        int[][] matrizB = new int[filas][columnas];
        int[][] matrizSuma = new int[filas][columnas];
        
        System.out.println("\nIngrese los elementos de la primera matriz:");
        ingresarMatriz(matrizA, scanner);
        
        System.out.println("\nIngrese los elementos de la segunda matriz:");
        ingresarMatriz(matrizB, scanner);
        
        sumarMatrices(matrizA, matrizB, matrizSuma);
        
        System.out.println("\n=== RESULTADOS ===");
        
        System.out.println("Matriz A:");
        mostrarMatriz(matrizA);
        
        System.out.println("Matriz B:");
        mostrarMatriz(matrizB);
        
        System.out.println("Suma (A + B):");
        mostrarMatriz(matrizSuma);
        
        scanner.close();
    }
    
    public static void ingresarMatriz(int[][] matriz, Scanner scanner) {
        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[0].length; j++) {
                System.out.print("Elemento [" + i + "][" + j + "]: ");
                matriz[i][j] = scanner.nextInt();
            }
        }
    }
    
    public static void sumarMatrices(int[][] matrizA, int[][] matrizB, int[][] matrizSuma) {
        for (int i = 0; i < matrizA.length; i++) {
            for (int j = 0; j < matrizA[0].length; j++) {
                matrizSuma[i][j] = matrizA[i][j] + matrizB[i][j];
            }
        }
    }
    
    public static void mostrarMatriz(int[][] matriz) {
        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[0].length; j++) {
                System.out.print(matriz[i][j] + "\t");
            }
            System.out.println();
        }
        System.out.println();
    }
}

File: SumaMatrices.c
#include <stdio.h>
#include <stdlib.h>

void ingresarMatriz(int **matriz, int filas, int columnas) {
    for (int i = 0; i < filas; i++) {
        for (int j = 0; j < columnas; j++) {
            printf("Elemento [%d][%d]: ", i, j);
            scanf("%d", &matriz[i][j]);
        }
    }
}

void sumarMatrices(int **matrizA, int **matrizB, int **matrizSuma, int filas, int columnas) {
    for (int i = 0; i < filas; i++) {
        for (int j = 0; j < columnas; j++) {
            matrizSuma[i][j] = matrizA[i][j] + matrizB[i][j];
        }
    }
}

void mostrarMatriz(int **matriz, int filas, int columnas) {
    for (int i = 0; i < filas; i++) {
        for (int j = 0; j < columnas; j++) {
            printf("%d\t", matriz[i][j]);
        }
        printf("\n");
    }
    printf("\n");
}

int main() {
    int filas, columnas;
    
    printf("Ingrese el número de filas: ");
    scanf("%d", &filas);
    printf("Ingrese el número de columnas: ");
    scanf("%d", &columnas);
    
    int **matrizA = (int **)malloc(filas * sizeof(int *));
    int **matrizB = (int **)malloc(filas * sizeof(int *));
    int **matrizSuma = (int **)malloc(filas * sizeof(int *));
    
    for (int i = 0; i < filas; i++) {
        matrizA[i] = (int *)malloc(columnas * sizeof(int));
        matrizB[i] = (int *)malloc(columnas * sizeof(int));
        matrizSuma[i] = (int *)malloc(columnas * sizeof(int));
    }
    
    printf("\nIngrese los elementos de la primera matriz:\n");
    ingresarMatriz(matrizA, filas, columnas);
    
    printf("\nIngrese los elementos de la segunda matriz:\n");
    ingresarMatriz(matrizB, filas, columnas);
    
    sumarMatrices(matrizA, matrizB, matrizSuma, filas, columnas);
    
    printf("\n=== RESULTADOS ===\n");
    printf("Matriz A:\n");
    mostrarMatriz(matrizA, filas, columnas);
    
    printf("Matriz B:\n");
    mostrarMatriz(matrizB, filas, columnas);
    
    printf("Suma (A + B):\n");
    mostrarMatriz(matrizSuma, filas, columnas);
    

    for (int i = 0; i < filas; i++) {
        free(matrizA[i]);
        free(matrizB[i]);
        free(matrizSuma[i]);
    }
    free(matrizA);
    free(matrizB);
    free(matrizSuma);
    
    return 0;
}